### ISSUE TITLE
Laravel built in casts now have incorrect type documented

### ISSUE THREAD

**Issue by Jefemy** [barryvdh/laravel-ide-helper#1423]
### Versions:
- ide-helper Version: 2.13.0
- Laravel Version: 9.52.0 
- PHP Version: 8.1.16

### Description:

Using Laravel built in casts such as `Illuminate\Database\Eloquent\Casts\AsCollection` will now return it as its own class, which doesn't actually implement a Collection meaning it will error out even when using it properly. Previously it returned as mixed which at least didn't error. The change causing this is #1388 but is more down to the [actual Laravel class](https://github.com/laravel/framework/blob/10.x/src/Illuminate/Database/Eloquent/Casts/AsCollection.php) not having type hints for the get method. This could be fixed on the Laravel side, or it could be extracted from the docblock return from the `castUsing` function. I did test with making a modification here but I got stuck as I believe the `CastsAttributes` wrapper in the docblock would need to be stripped for proper functionality and I wasn't quite sure on the best way to do that

### Steps To Reproduce:

- Create a Model like this
```php
class CollectionModel extends Model
{
    public $casts = [
        'collection' => \Illuminate\Database\Eloquent\Casts\AsCollection::class,
    ];
}
```

- Generate model docblocks `php artisan ide-helper:model -M`
- Outputted type is `@property \Illuminate\Database\Eloquent\Casts\AsCollection $collection`

**Comment by Jefemy** [comment]
I am now seeing that #1393 exists but it is also missing types for the 5 other casts that Laravel has. If that turns out to be the proper way to solve this I can make a pull containing all of them.

**Comment by jamesrusso** [comment]
I'm running into this same issue.

**Comment by moussaclarke** [comment]
@jamesrusso My workaround is to keep generated model types in separate file and manually over-ride broken types only in the relevant model files.

**Comment by barryvdh** [comment]
Fixed with #1393?
